src/extension/features/accounts/calculate-irr/index.js

```diff
@@ -0,0 +1,230 @@
+    let transactions = account.getTransactions();
+
+    if (
+      transactions.reduce((reduced, transaction) => {
+        if (transaction.flag === this._flagColor && transaction.cleared !== 'Uncleared') {
+          return reduced + transaction.amount;
+        }
+        return reduced;
+      }, 0) === 0
+    )
+      return Infinity;
+
+    var fromDate = transactions
+      .reduce((date, transaction) => {
+        if (transaction.date.isBefore(date)) return transaction.date;
+        return date;
+      }, transactions[0].date)
+      .setDate(1)
+      .setMonth(0);
+    var toDate = transactions.reduce((date, transaction) => {
+      if (transaction.date.isAfter(date)) return transaction.date;
+      return date;
+    }, transactions[0].date);
+
+    if (options && options.fromDate && fromDate.isBefore(options.fromDate)) {
+      fromDate = options.fromDate;
+    }
+    if (options && options.toDate && toDate.isAfter(options.toDate)) {
+      toDate = options.toDate;
+    }
+
+    let startingValue = transactions.reduce((reduced, transaction) => {
+      if (
+        transaction.cleared !== 'Uncleared' &&
+        !transaction.isTombstone &&
+        transaction.date.isBefore(fromDate)
+      ) {
+        return reduced + transaction.amount;
+      }
+      return reduced;
+    }, 0);
+
+    let currentValue = transactions.reduce((reduced, transaction) => {
+      if (
+        transaction.cleared !== 'Uncleared' &&
+        !transaction.isTombstone &&
+        (transaction.date.isBefore(toDate) || transaction.date.equalsDateWithoutTime(toDate))
+      ) {
+        return reduced + transaction.amount;
+      }
+      return reduced;
+    }, 0);
+
+    let contributions = transactions.filter(trans => {
+      return (
+        trans.flag === this._flagColor &&
+        trans.cleared !== 'Uncleared' &&
+        (trans.date.isAfter(fromDate) || trans.date.equalsDateWithoutTime(fromDate)) &&
+        (trans.date.isBefore(toDate) || trans.date.equalsDateWithoutTime(toDate))
+      );
+    });
+
+    var resultRate = 0.1;
+    var newRate;
+    var epsRate;
+    var resultValue;
+    var iteration = 0;
+    var contLoop = true;
+    do {
+      resultValue = this._irrResult({
+        startingValue,
+        fromDate,
+        transaction: contributions,
+        rate: resultRate,
+        currentValue,
+        toDate,
+      });
+      var deriv = this._irrFirstDeriv({
+        startingValue,
+        fromDate,
+        transaction: contributions,
+        rate: resultRate,
+        currentValue,
+        toDate,
+      });
+      newRate = resultRate - resultValue / deriv;
+      epsRate = Math.abs(newRate - resultRate);
+      resultRate = newRate;
+      contLoop = epsRate > epsMax && Math.abs(resultValue) > epsMax;
+    } while (contLoop && ++iteration < iterMax);
+
+    if (contLoop) {
+      console.log('Max Iterations Exceeded');
+      return NaN;
+    }
+
+    return resultRate;
+  };
+
+  _irrResult = ob => {
```

Comment:
I'm fine with abbreviations in `css` but in code that makes it a lot harder to understand what we're doing. in the spirit of my previous comment and making sure this code stands the test of time as issues inevitably get debugged, let's try to use full words for variables.